src/analysis/import/import_jobs.sh

```diff
@@ -45,29 +45,44 @@ function import_job_data() {
 
     NB_STATE_ABBREV="${1}"
     NB_DATA_TYPE="${2:-main}"    # Either 'main' or 'aux'
+
     NB_JOB_FILENAME="${NB_STATE_ABBREV}_od_${NB_DATA_TYPE}_JT00_2017.csv"
+    S3_PATH="s3://${AWS_STORAGE_BUCKET_NAME}/data/${NB_JOB_FILENAME}.gz"
 
     if [ -f "/data/${NB_JOB_FILENAME}.gz" ]; then
         JOB_DOWNLOAD="/data/${NB_JOB_FILENAME}.gz"
-    elif [ "${AWS_STORAGE_BUCKET_NAME}" ] && aws s3 ls "s3://${AWS_STORAGE_BUCKET_NAME}/data/${NB_JOB_FILENAME}.gz"; then
+        echo "Using local job data file ${JOB_DOWNLOAD}"
+    elif [ "${AWS_STORAGE_BUCKET_NAME}" ] && aws s3 ls "${S3_PATH}"; then
         JOB_DOWNLOAD="${NB_TEMPDIR}/${NB_JOB_FILENAME}.gz"
-        aws s3 cp "s3://${AWS_STORAGE_BUCKET_NAME}/data/${NB_JOB_FILENAME}.gz" "${JOB_DOWNLOAD}"
+        aws s3 cp "${S3_PATH}" "${JOB_DOWNLOAD}"
+        echo "Downloaded job data file ${JOB_DOWNLOAD} from S3"
     else
         JOB_DOWNLOAD="${NB_TEMPDIR}/${NB_JOB_FILENAME}.gz"
         set +e
         wget -nv -O "${JOB_DOWNLOAD}" "http://lehd.ces.census.gov/data/lodes/LODES7/${NB_STATE_ABBREV}/od/${NB_JOB_FILENAME}.gz"
         WGET_STATUS=$?
         set -e
+        # If the 2017 file isn't there (SD and AK), do the check/download/cache again for 2016
         if [[ $WGET_STATUS -eq 8 ]]; then
             echo "No 2017 job data available, falling back to 2016 data..."
-            JOB_DOWNLOAD="${NB_TEMPDIR}/${NB_JOB_FILENAME}.gz"
             NB_JOB_FILENAME="${NB_STATE_ABBREV}_od_${NB_DATA_TYPE}_JT00_2016.csv"
-            wget -nv -O "${JOB_DOWNLOAD}" "http://lehd.ces.census.gov/data/lodes/LODES7/${NB_STATE_ABBREV}/od/${NB_JOB_FILENAME}.gz"
+            S3_PATH="s3://${AWS_STORAGE_BUCKET_NAME}/data/${NB_JOB_FILENAME}.gz"
+            JOB_DOWNLOAD="${NB_TEMPDIR}/${NB_JOB_FILENAME}.gz"
+            if [ "${AWS_STORAGE_BUCKET_NAME}" ] && aws s3 ls "${S3_PATH}"; then
+                aws s3 cp "${S3_PATH}" "${JOB_DOWNLOAD}"
+                echo "Downloaded job data file from S3"
+            else
+                wget -nv -O "${JOB_DOWNLOAD}" "http://lehd.ces.census.gov/data/lodes/LODES7/${NB_STATE_ABBREV}/od/${NB_JOB_FILENAME}.gz"
+                echo "Uploading job data file to S3 cache"
+                aws s3 cp "${JOB_DOWNLOAD}" "${S3_PATH}"
```

Comment:
I flailed a bit remembering how to run local analyses, which caused me to do a run of this where `AWS_STORAGE_BUCKET_NAME` was an empty string, and I ended up with an error at this point because I hadn't provided it with a bucket to upload to (`S3_PATH` was `s3:///data/...`). The Analysis README mentions that `AWS_STORAGE BUCKET' will be "Ignored unless `PFB_JOB_ID` and `AWS_PROFILE` are set," and this doesn't seem to be quite doing that because it's causing a failure if unset. But I'm not sure how relevant or necessary it is to be able to run analyses without enabling this functionality, or whether in practice anyone ever runs without those two variables set, but I wanted to bring it up in case this needs to be put inside a guard of some kind.

Reply:
Good point. I know PFB runs local analyses, though I'm not sure whether they set `AWS_STORAGE_BUCKET_NAME` or not.  In any case, we definitely don't want to break that use case. I added conditions.